shell/common/platform_util_linux.cc

```diff
@@ -122,6 +122,19 @@ bool XDGUtil(const std::vector<std::string>& argv,
   // bring up a new terminal if necessary.  See "man mailcap".
   options.environment["MM_NOTTTY"] = "1";
 
+  // This is set by Chromium here:
+  // https://chromium-review.googlesource.com/c/chromium/src/+/2586184
+  // and can detrimentally affect external app behaviors, so we want to
+  // unset it and replace with the user's setting if one exists.
+  std::string gdk_backend;
+  if (base::Environment::Create()->GetVar("GDK_BACKEND", &gdk_backend)) {
```

Comment:
This will only get the value set by chromium, as `base::Environment` uses `[get/set]env` api internally. We should cache the value of `GDK_BACKEND` before we call `BuildGtkUi` https://github.com/electron/electron/blob/c37c9adbe2fc457d02396f4c02ac536a60a080ca/shell/browser/electron_browser_main_parts.cc#L380, so that we get expected value set by the user.